login/service.go

```diff
@@ -191,3 +192,27 @@ type ghUser struct {
 	Name      string `json:"name"`
 	AvatarURL string `json:"avatar_url"`
 }
+
+// ServiceAsUser creates a new service and fill the context with input user
+func ServiceAsUser(serviceName string, u account.User) *goa.Service {
+	svc := goa.New(serviceName)
+	svc.Context = WithIdentity(svc.Context, u.ID)
+	return svc
+}
+
+//CurrentUserInContextKey is a key that will be used to put and to get `currentUser` from goa.context
+const CurrentUserInContextKey = "currentUser"
+
+// WithIdentity fills the context with input UUID
+func WithIdentity(ctx context.Context, id uuid.UUID) context.Context {
```

Comment:
Yes, got the point. Changed in https://github.com/almighty/almighty-core/pull/387/commits/22e3b1ce34e3d70851ce2ea74f7ced2740391b4e

On similar line, should input be `interface{}` so that only setter knows how to get the value (datatype wise)